packages/edit-site/src/components/page-pages/index.js

```diff
@@ -26,13 +26,23 @@ const EMPTY_OBJECT = {};
 export default function PagePages() {
 	const [ view, setView ] = useState( {
 		type: 'list',
-		search: '',
+		filters: {
+			search: '',
+			status: 'publish, draft',
+		},
 		page: 1,
 		perPage: 5,
 		sort: {
 			field: 'date',
 			direction: 'desc',
 		},
+		layout: {
+			filters: [
```

Comment:
For the `DataView` component to be generic and reusable in different contexts, we need to provide the filter metadata dynamically. This also includes the `search` filter, as there may be endpoints that don't have it, etc.

In this PR the filters are always visible, but the design direction is that the user should be able to add/remove them from the UI. If/When this happens, we need to serialize the filter metadata as part of the view anyway.

An alternative I thought was to declare this information in the fields instead. There are two things that make me think `filters<=>fields` is not a clear mapping:

- UI order: how do make sure the `search` filter goes first, and then `author`, and finally `status`? We could have a `order` property for the filter, sure, though that approach starts to get convoluted as soon as you have many filters. By storing them in the view, we already have them ordered – even more, in the future, users could change their position (drag&drop, when they create a new view/page, etc.).
- Filters can be cross-fields. For example, for the pages endpoint, the `search` is a filter that affects `post_title`, `post_except`, and `post_content`. Or only some of them if `search_columns` is provided. How do we connect that to fields? One option would be to have an invisible `search` field that declares the filter metadata, though it starts to stretch a bit the `filter<=>field` relationship in my view.

Note that the issues are solvable and we could start with any approach. I just thought that having the filter metadata as part of the `view` was conceptually clearer, and enables us to serialize the views automatically.

Reply:
The view data structure is going to be persisted on the database. I think ideally we would not be persisting the metadata that says an author is an enumeration filter in the database that is something static and that we don't expect the user to have control over. I would be ok with having something like filterInfo like we have paginationInfo outside the view for now.

Reply:
Regarding the content of the structure, I would expect author and status filter types to be automatically generated based on the field type and the information author, and status is an enumeration. For more complex filters like search or other filters that affect multiple fields, we could still programmatically declare them on a filter structure. I think we can have the filter structure but have a function that based on the field automatically generates some filter types in that structure. We can leave that for a follow-up.

Reply:
> I think ideally we would not be persisting the metadata that says an author is an enumeration filter in the database that is something static and that we don't expect the user to have control over

This is great feedback, thanks. I'll address this in a bit.

Reply:
I've separated `view.layout.filters` into:

- **view.visibleFilters**: the view config lists the visible filters and their ordering. This is what we serialize. Example: `[ 'search', 'author', 'status' ]`.
- **field.filters**: the field config registers the available filters. Example: `[ { type: 'search', id: 'search' }, { type: 'enumeration', id: 'author' } ]`.